### PR TITLE
move DataRecall and DataLock to separate file

### PR DESCRIPTION
**Pull request by belforte**
This is a first step toward https://github.com/dmwm/CRABServer/issues/7733
At this time I'd like to have feedback on the separation of code in two files, the class definition/use etc.

Maybe a single class should be better ? I am not very happy with two classes which only differ on the name of the Rucio account they use internally. But this is the first time I create classes, with inheritance as well.. so I am sure I did in a way which could be vastly improved.

This is supposed not to change functionality wrt what is in v3.230712 and I have done some limited testing using the internal `main()` method. 

Once we have settled on the design, I will tackle remaining pylint issues, which looks straighforward.

Thanks !

### CODE REVIEW

**Review by novicecpp — COMMENTED** [review]
I am leaning towards a single class.

From the caller's point of view, it is indeed `RucioAction` for both `recallData` and `lockData`, and we can easily initialize the class with a different account and call the class method we want independently.
I separate my RUCIO_Transfers.py's action into three classes because it is easier to manage completely different actions (Well, it can be a simple function, but it is cleaner to use `self.logger` and init logger in `__init__()`).

**Comment by novicecpp** [comment]
Your changes in general look really nice and clean. I like it a lot!

**Comment by belforte** [comment]
OK. I will reduce to one class. I do not like all those `__init__` (and did not find the way to use `super`)
Thanks !

**Comment by belforte** [comment]
DBSDataDiscovery is very good pylint-wise now.
But since I moved one constant to ServerUtilities, there are tons of complaints from that.
WIll have a quick look

**Comment by belforte** [comment]
Can not do much about ServerUtilities because it needs to run in Client too (python2).

We should separate in a different file things common to client. Then it will be safe to touch.

**Comment by belforte** [comment]
DBSDataDIscovery is now down to one only pylint warngin, which I do not know how to solve !

**Comment by belforte** [comment]
time to rebase and squash

**Comment by belforte** [comment]
Now I need to test for good. And possibly close this and make a new PR once it works and with a simpler commit history.
Anyhow review for "proper designa and style" can be done.
I know I have a lot of spaces to fix for pep-8 in DBDDataDiscovery. Will do.

**Comment by belforte** [comment]
Jenkins tests
- [ ] #7763
- [ ] #7764

**Comment by belforte** [comment]
No harm to standard functionalities. Will make a separate ad-hoc test for tapeRecall

File: src/python/TaskWorker/Actions/RucioActions.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    Allows for different actions to use different Rucio accounts
+    """
+    def __init__(self, config=None, crabserver=None, rucioAcccount=None,
+                 taskName=None, username=None, logger=None):
+        self.config = config
+        self.crabserver = crabserver
+        self.taskName = taskName
+        self.logger = logger
+        self.username = username
+        # following two will be filled in inherited subclasses TapeRecaller and InputLocker
```
Comment (novicecpp) [inline src/python/TaskWorker/Actions/RucioActions.py:31]: Do not forget to remove this line :)

**Review by novicecpp — APPROVED** [review]
LGTM. 
Thanks Stefano.

File: src/python/TaskWorker/Actions/RucioActions.py (hunk 1 above)
Comment (belforte) [reply src/python/TaskWorker/Actions/RucioActions.py]: Thanks Wa !!!

**Pull request merged**